app/app.go

```diff
@@ -1041,16 +1041,28 @@ func New(
 	}
 
 	if loadLatest {
-		if err := app.LoadLatestVersion(); err != nil {
-			tmos.Exit(err.Error())
+		var qmsVersion int64
+		if app.qms != nil {
+			qmsVersion = app.qms.LatestVersion()
 		}
 
-		if app.qms != nil {
-			v1 := app.qms.LatestVersion()
-			v2 := app.LastBlockHeight()
-			if v1 > 0 && v1 < v2 {
+		if qmsVersion == 0 {
+			if err := app.LoadLatestVersion(); err != nil {
+				tmos.Exit(err.Error())
+			}
+		} else {
+			// make sure iavl version is not ahead of versiondb version
+			app.SetStoreLoader(VersionStoreLoader(qmsVersion))
```

Comment:
seems storeLoader in [upgrade](https://github.com/crypto-org-chain/cronos/blob/main/app/upgrades.go#L57) get overwritten

Reply:
good catch, fixed, please review again.